assets/wizards/checklist/index.js

```diff
@@ -27,6 +27,27 @@ class ChecklistScreen extends Component {
 		};
 	}
 
+	/**
+	 * When the checklist is first loaded, set the progress.
+	 * This counts from the first element until the first non-complete element to determine progress.
+	 * This will prevent issues if e.g. a wizard further down the list is complete.
+	 */
+	componentDidMount() {
+		const { steps } = this.props;
+		let numComplete = 0;
+		for ( let i = 0; i < steps.length; ++i ) {
+			if ( steps[i].completed ) {
+				++numComplete;
+			} else {
+				break;
+			}
+		}
+
+		this.setState( {
+			checklistProgress: numComplete,
+		} );
```

Comment:
[Suggestion] You could condense this a bit with Array.reduce, like this:
```
const { steps } = this.props;
		this.setState( {
			checklistProgress: steps.reduce(
				( accumulator, step ) => accumulator + ( step.completed ? 1 : 0 ),
				0
			),
		} );
```

Reply:
Thanks for the suggestion. The reason I didn't go that direction is because if e.g. the 1st, 2nd, and 4th wizards are complete it would have checklistProgress as `3`, instead of the more-correct `2`.